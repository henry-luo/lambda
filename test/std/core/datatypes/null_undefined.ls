// Test: Null and Undefined Values
// Category: core/datatypes
// Type: positive
// Expected: 1

// Null and undefined values
let n = null
n
let u = undefined
u
let eq_null = n == null      
eq_null
let eq_undefined = u == null 
eq_undefined
let strict_eq = n === null   
strict_eq
let strict_neq = n === u     
strict_neq
let is_null = n === null     
is_null
let is_undefined = u === undefined 
is_undefined
let default1 = n ?? "default"  
default1
let default2 = u ?? 42         
default2
let default3 = 0 ?? "default"  
default3
let obj = {a: {b: 42}}
obj
let value1 = obj?.a?.b     
value1
let value2 = obj?.x?.y     
value2
1
