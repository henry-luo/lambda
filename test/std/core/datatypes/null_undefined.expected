09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/null_undefined.ls...
09:15:42 [ERROR] Syntax tree has errors.
09:15:42 [ERROR] Syntax error at Ln 13, Col 21 - 13, Col 22: ERROR
09:15:42 [ERROR] ERROR node at Ln 13, Col 21 - 13, Col 22
09:15:42 [ERROR] Syntax error at Ln 14, Col 22 - 14, Col 23: ERROR
09:15:42 [ERROR] ERROR node at Ln 14, Col 22 - 14, Col 23
09:15:42 [ERROR] Syntax error at Ln 17, Col 19 - 17, Col 20: ERROR
09:15:42 [ERROR] ERROR node at Ln 17, Col 19 - 17, Col 20
09:15:42 [ERROR] Syntax error at Ln 18, Col 24 - 18, Col 25: ERROR
09:15:42 [ERROR] ERROR node at Ln 18, Col 24 - 18, Col 25
09:15:42 [ERROR] Syntax error at Ln 21, Col 16 - 21, Col 20: ERROR
09:15:42 [ERROR] ERROR node at Ln 21, Col 16 - 21, Col 20
09:15:42 [ERROR] Syntax error at Ln 22, Col 16 - 22, Col 20: ERROR
09:15:42 [ERROR] ERROR node at Ln 22, Col 16 - 22, Col 20
09:15:42 [ERROR] Syntax error at Ln 23, Col 16 - 23, Col 20: ERROR
09:15:42 [ERROR] ERROR node at Ln 23, Col 16 - 23, Col 20
09:15:42 [ERROR] Syntax error at Ln 27, Col 17 - 27, Col 18: ERROR
09:15:42 [ERROR] ERROR node at Ln 27, Col 17 - 27, Col 18
09:15:42 [ERROR] Syntax error at Ln 27, Col 20 - 27, Col 21: ERROR
09:15:42 [ERROR] ERROR node at Ln 27, Col 20 - 27, Col 21
09:15:42 [ERROR] Syntax error at Ln 28, Col 17 - 28, Col 18: ERROR
09:15:42 [ERROR] ERROR node at Ln 28, Col 17 - 28, Col 18
09:15:42 [ERROR] Syntax error at Ln 28, Col 20 - 28, Col 21: ERROR
09:15:42 [ERROR] ERROR node at Ln 28, Col 20 - 28, Col 21
09:15:42 [ERROR] Error: Failed to compile the function.
##### Script 'test/std/core/datatypes/null_undefined.ls' executed: #####
null
